contracts/contracts/core/GasTankManager.sol

```diff
@@ -0,0 +1,113 @@
+import {IGasTankManager} from "../interfaces/IGasTankManager.sol";
+import {GasTankManagerStorage} from "./GasTankManagerStorage.sol";
+import {Errors} from "../utils/Errors.sol";
+
+/// @title GasTankManager
+/// @notice Coordinates on-demand ETH bridging so EOA gas tank balances stay funded for mev-commit transactions.
+/// @dev The RPC provider manages EOA gas tank balances on the Mev Commit chain.
+/// @dev Flow overview:
+/// - EOA (L1)
+///   - Authorizes and sets this contract as a delegate against its own EOA address. (ERC-7702)
+///   - Calls `sendMinimumDeposit` to send the initial ETH funds to their gas tank on the Mev Commit chain.
+/// - Provider (mev-commit)
+///   - Triggers a top-up via `topUpGasTank` when the gas tank balance cannot cover the next mev-commit transaction. This transfer amount is always the difference
+///   between the `minDeposit` and the current balance of this contract. The provider is the only one who can trigger a top-up.
+///   - These funds are then transferred to the EOA gas tank on the Mev Commit chain.
+///   - When a mev-commit transaction is made, the provider deducts the amount needed from the EOA's gas tank.
+contract GasTankManager is IGasTankManager, GasTankManagerStorage, Ownable2StepUpgradeable, UUPSUpgradeable {
+    /// @notice Restricts calls to those triggered internally via `onlyThisEOA`.
+    modifier onlyThisEOA() {
+        require(msg.sender == address(this), NotThisEOA(msg.sender, address(this)));
+        _;
+    }
+
+    /// @notice Locks the implementation upon deployment.
+    /// @dev See https://docs.openzeppelin.com/upgrades-plugins/writing-upgradeable#initializing-the-implementation-contract
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Accepts direct ETH deposits and forwards them to the provider.
+    receive() external payable {
+        _sendFundsToProvider(msg.value);
+    }
+
+    /// @notice Reverts on any call data to keep the interface surface narrow.
+    fallback() external payable {
+        revert Errors.InvalidFallback();
+    }
+
+    /// @notice Initializes ownership and baseline deposit requirement.
+    /// @param _owner EOA managed by the RPC provider.
+    /// @param _minDeposit Initial deposit requirement in wei.
+    function initialize(address _owner, uint256 _minDeposit) external initializer {
+        minDeposit = _minDeposit;
+        __Ownable_init(_owner);
+        __UUPSUpgradeable_init();
+    }
+
+    /// @inheritdoc IGasTankManager
+    function setMinimumDeposit(uint256 _minDeposit) external onlyOwner {
+        minDeposit = _minDeposit;
+        emit MinimumDepositSet(_minDeposit);
+    }
+
+    /// @inheritdoc IGasTankManager
+    function topUpGasTank(uint256 _gasTankBalance) external onlyOwner {
+        uint256 minDeposit_ = minDeposit;
+        uint256 available = address(this).balance;
```

Comment:
As we talked about. we will on be tracking the balance on chain. This will be off-chain.

Reply:
This was never storing the balance on-chain. It's purpose was to determine the exact amount needed for a transfer that would equal the `minDeposit`. It was initially done this way to prevent transferring more funds then actually required. If the RPC is going to wait this the gas tank funds are exhausted, then simply passing the `minDeposit` amount each time makes sense. I have changed it to do exactly that.